src/init.cpp

```diff
@@ -1026,6 +1027,35 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
         }
     }
 
+    if (!mapMultiArgs["-buriedsfparams"].empty()) {
+        // Allow overriding ism parameters for testing
+        if (!Params().MineBlocksOnDemand()) {
```

Comment:
Instead of abusing MineBlocksOnDemand(), maybe a new attribute in CChainParams?
What about fAllowOverwriteBuriedSoftforks?
We don't even need a new getter, CChainParams is accessed as a const ref everywhere outside chainparams.cpp.